modules/viewHierarchyMixin.js

```diff
@@ -106,49 +111,201 @@
      * @method cleanupSelf
      */
     cleanupSelf: function() {
+      this.deactivate();
+
       // Clean up child views first
       this.cleanupChildViews();
 
+      // Remove view from DOM
+      this.remove();
+
       // Unbind all local event bindings
       this.unbind();
       this.off();
       this.stopListening();
 
-      // Remove view from DOM
-      this.remove();
-
-      // Undelegates events
-      this.undelegateEvents();
-
       // Delete the dom references
       delete this.$el;
       delete this.el;
     },
 
     /**
+     * Method to be invoked when deactivate is called. Use this method to turn off any
+     * custom timers, listenTo's or on's that should be deactivatable.
+     * @method deactivateCallback
+     */
+    deactivateCallback: function() {
+      // do nothing
+    },
+
+    /**
+     * Method to be invoked when activate is called. Use this method to turn on any
+     * custom timers, listenTo's or on's that should be activatable.
+     * @method deactivateCallback
+     */
+    activateCallback: function() {
+      // do nothing
+    },
+
+    /**
      * Default child view cleanup method that may be overriden.
      * @method cleanupChildViews
      */
     cleanupChildViews: function() {
-      // do nothing
+      _.each(this._childViews, function(view) {
+        view.dispose();
+      }, this);
```

Comment:
what is the this parameter at the end of the _.each method call?  Do you need it?